ui/components/app/collectible-details/index.scss

```diff
@@ -53,18 +53,41 @@ $spacer-break-small: 16px;
     overflow-wrap: break-word;
   }
 
-  &__contract-link,
-  &__image-link {
-    word-break: break-all;
+  &__contract-wrapper {
+    max-width: calc(100% - #{$link-title-width});
+  }
+
+  &__contract-copy-button {
+    @include H6;
 
-    @media screen and (max-width: $break-small) {
-      overflow: hidden;
-      text-overflow: ellipsis;
-      white-space: nowrap;
-      width: 90%;
+    width: 80px;
+    display: flex;
+    align-items: flex-start;
+    justify-content: center;
+    background-color: transparent;
+    cursor: pointer;
+    color: var(--ui-4);
+    border: 0;
+
+    &:active {
+      transform: scale(0.97);
     }
   }
 
+  &__contract-link {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+    max-width: 90%;
```

Comment:
Seemingly random width percentages raise ui red flags for me. Also not sure this is needed as the parent box element is dictating the width here. So it should still as remain as text ellipses if this is removed.

```suggestion
```

<img width="1398" alt="Screen Shot 2022-01-26 at 12 21 34 PM" src="https://user-images.githubusercontent.com/8112138/151076376-85ed2779-26cf-4f7a-b521-cd15cfbd4a54.png">

Reply:
Agreed on this one!

Reply:
removed here: https://github.com/MetaMask/metamask-extension/pull/13388/commits/240003db3370d7ee2ec01f1b43548c38f6f76bb6